### PR TITLE
Remove `get_gpu_memory_map` deprecated since v1.5

### PR DESCRIPTION
**Pull request by akihironitta**
## What does this PR do?
Part of #15618.
The API was deprecated in #9921.

### Does your PR introduce any breaking changes? If yes, please list them.
Removes
```python
pytorch_lightning.utilities.memory.get_gpu_memory_map
```
in favor of
```python
pytorch_lightning.accelerators.cuda.get_nvidia_gpu_stats
```

## Before submitting

- [x] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/Lightning-AI/lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [x] Did you make sure to **update the documentation** with your changes? (if necessary)
- [x] Did you write any **new necessary tests**? (not for typos and docs)
- [x] Did you verify new and **existing tests pass** locally with your changes?
- [x] Did you list all the **breaking changes** introduced by this pull request?
- [x] Did you **update the CHANGELOG**? (not for typos, docs, test updates, or minor internal changes/refactors)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing, make sure you have read the [review guidelines](https://github.com/Lightning-AI/lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

cc @borda @justusschock @awaelchli @akihironitta

### CODE REVIEW

**Comment by akihironitta** [comment]
Not sure which milestone this change is part of, v1.8.x or v1.9.

File: src/pytorch_lightning/utilities/memory.py [hunk 1]
```diff
@@ -96,38 +96,6 @@ def garbage_collection_cuda() -> None:
             raise
 
 
-def get_gpu_memory_map() -> Dict[str, float]:
```
Comment (awaelchli) [inline src/pytorch_lightning/utilities/memory.py:99]: Good find @akihironitta! 
Do you think we should add it to the `src/pytorch_lightning/_graveyard`? cc @carmocca

**Review by awaelchli — APPROVED** [review]

**Comment by awaelchli** [comment]
> Not sure which milestone this change is part of, v1.8.x or v1.9.

Removals of deprecated API always goes to milestone 1.X.0 :)

File: src/pytorch_lightning/utilities/memory.py (hunk 1 above)
Comment (akihironitta) [reply src/pytorch_lightning/utilities/memory.py:99]: This is something @Borda found actually :)

I hope I did this right in 13afae3 as this is my first time burying something in the graveyard :)

**Review by Borda — APPROVED** [review]

**Comment by Borda** [comment]
> > Not sure which milestone this change is part of, v1.8.x or v1.9.
> 
> Removals of deprecated API always goes to milestone 1.X.0 :)

On the other hand this is supposed to be removed already fro 1.7.0 :innocent:

File: src/pytorch_lightning/_graveyard/utilities.py [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import sys
+from typing import Any
+
+
+def _patch_sys_modules() -> None:
+    # TODO: Remove in v2.0.0
+    self = sys.modules[__name__]
```
Comment (carmocca) [inline src/pytorch_lightning/_graveyard/utilities.py:21]: This is only done if the module does not exist, but in this case it does.

You just need to monkeypatch the attribute to the existing one

I'll help

Comment (akihironitta) [reply src/pytorch_lightning/_graveyard/utilities.py]: Done! Thank you for the guidance!

File: src/pytorch_lightning/_graveyard/__init__.py [hunk 3]
```diff
@@ -17,4 +17,5 @@
 import pytorch_lightning._graveyard.legacy_import_unpickler
 import pytorch_lightning._graveyard.loggers
 import pytorch_lightning._graveyard.trainer
-import pytorch_lightning._graveyard.training_type  # noqa: F401
+import pytorch_lightning._graveyard.training_type
+import pytorch_lightning._graveyard.utilities
```
Comment (carmocca) [inline src/pytorch_lightning/_graveyard/__init__.py:21]: pre-commit has several failures, one of them being that you need to keep the noqa in the last line

**Review by carmocca — APPROVED** [review]

**Pull request merged**